[GTK][WPE] Build broken with ENABLE_WEB_AUDIO=OFF

blino opened the pull request:
#### 7ffe7f5c6b9c96f18abdd1bfefbb145de30fe3b9
<pre>
[GTK][WPE] Build broken with ENABLE_WEB_AUDIO=OFF
<a href="https://bugs.webkit.org/show_bug.cgi?id=281159">https://bugs.webkit.org/show_bug.cgi?id=281159</a>

Reviewed by Miguel Gomez.

Since 280453@main, default values from UnifiedWebPreferences.yaml are used.
Do not try to use WebAudioEnabled when undefined.

* Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
(webkit_settings_class_init):

Canonical link: <a href="https://commits.webkit.org/284970@main">https://commits.webkit.org/284970@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5ba05e3709159e356ccbca317399b351ae8ae4e5

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/71047 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/50459 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/23820 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/75151 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/22252 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/58258 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/22070 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/56175 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/14649 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/74113 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/45845 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/61242 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/36617 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/42494 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/20593 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/64427 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/19044 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/76871 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/15279 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/18229 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/63919 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/15323 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/61316 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/63869 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/15730 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/11985 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/5622 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/46260 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/1036 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/47332 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/48615 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/47074 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

philn commented on Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
```diff
@@ -1105,7 +1105,11 @@ static void webkit_settings_class_init(WebKitSettingsClass* klass)
+#if ENABLE(WEB_AUDIO)
             FEATURE_DEFAULT(WebAudioEnabled),
+#else
+            TRUE,
```
shouldn't this be FALSE?

blino replied:
Like previous fixups around WebKitSettings since commit ff6ffc06d5f8dda289e1d51e3b9a86d3370f342b, I have reused the same value than before the UnifiedWebPreferences.yaml switch in this commit.

blino replied:
See also the default value for `enable-media`.

But maybe it would make more sense to default to `FALSE` there too if the feature is not enabeld.

philn replied:
cc @aperezdc

blino replied:
I have changed to `FALSE` for the WebAudio case

aperezdc replied:
I agree this should be `FALSE` because when WebAudio is disabled at build time it's not really possible to ever enable it 😉 

I would even argue that we should prevent changing the setting to `TRUE`, and emit a warning.

blino replied:
Thanks all for the feedback.
Ok to merge as-is for a first step?

magomez replied:
> Thanks all for the feedback. Ok to merge as-is for a first step?

go head!

blino committed:
[GTK][WPE] Build broken with ENABLE_WEB_AUDIO=OFF
https://bugs.webkit.org/show_bug.cgi?id=281159

Reviewed by Miguel Gomez.

Since 280453@main, default values from UnifiedWebPreferences.yaml are used.
Do not try to use WebAudioEnabled when undefined.

* Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
(webkit_settings_class_init):

Canonical link: https://commits.webkit.org/284970@main
